implementations/rust/ockam/ockam_command/src/policy/show.rs

```diff
@@ -33,10 +42,89 @@ impl ShowCommand {
     }
 
     async fn async_run(&self, ctx: &Context, opts: CommandGlobalOpts) -> miette::Result<()> {
-        let node = BackgroundNodeClient::create_to_node(ctx, &opts.state, &self.at).await?;
-        let req = Request::get(policy_path(&self.resource, &self.action));
-        let policy: Policy = node.ask(ctx, req).await?;
-        println!("{}", policy.expression());
+        ShowTui::run(ctx, opts, self.clone()).await
+    }
+}
+
+pub struct ShowTui {
+    ctx: Context,
+    opts: CommandGlobalOpts,
+    node: BackgroundNodeClient,
+    cmd: ShowCommand,
+    resource: ResourceTypeOrName,
+}
+
+impl ShowTui {
+    pub async fn run(
+        ctx: &Context,
+        opts: CommandGlobalOpts,
+        cmd: ShowCommand,
+    ) -> miette::Result<()> {
+        let node = BackgroundNodeClient::create(ctx, &opts.state, &cmd.at).await?;
+        let resource_type = if cmd.resource_type.is_none() && cmd.resource.is_none() {
```

Comment:
I haven't seen one example/test where we don't specify resource or resource type. I don't really see why having a default value matters here.

Reply:
The reason for that is pretty much the same as [here](https://github.com/build-trust/ockam/pull/7572#discussion_r1497222491). From the UX perspective, I think it's better to just show a policy rather than return an error). It's also how the rest of the `show` commands behave, so I didn't want to have a special behavior on this one.

Reply:
I see. I'm not sure I agree about the UX side. I think it makes sense for commands creating entities by default but not for policies. Maybe we can involve @mrinalwadhwa and @nazmulidris for their thoughts on this (but this doesn't have to hold up the PR).